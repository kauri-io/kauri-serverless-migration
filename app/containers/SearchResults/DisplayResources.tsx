import { Grid, Theme } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'
import { compose, graphql } from 'react-apollo'
import withLoading from '../../lib/with-loading'
import withPagination from '../../lib/with-pagination'
import { searchResultsAutocomplete } from '../../queries/Search'
import LoadingComponent from '../../components/Loading'
import PublicProfileCard from '../../components/Card/PublicProfileCard'
import CollectionCard from '../../components/Card/CollectionCard'
import CommunityCard from '../../components/Card/CommunityCard'
import ArticleCard from '../../components/Card/ArticleCard'

const useStyles = makeStyles((theme: Theme) => ({
    container: {
        padding: theme.spacing(0, 2, 2, 2),
        maxWidth: 808,
        margin: 'auto',
        '& > *': {
            margin: theme.spacing(1),
        },
    },
}))

const Search = ({ loading, data: { searchAutocomplete } }) => {
    const classes = useStyles({})

    if (loading) return <LoadingComponent />

    return (
        <Grid className={classes.container}>
            {searchAutocomplete &&
                searchAutocomplete.content &&
                searchAutocomplete.content.map(i => {
                    const resource = i.resource
                    const type =
                        i.resourceIdentifier && i.resourceIdentifier.type
                    switch (type) {
                        case 'ARTICLE':
                            return <ArticleCard {...resource} />
                        case 'COLLECTION':
                            return <CollectionCard {...resource} />
                        case 'COMMUNITY':
                            return <CommunityCard {...resource} />
                        case 'USER':
                            return <PublicProfileCard {...resource} />
                        default:
                            return null
                    }
                })}
        </Grid>
    )
}

export default compose(
    graphql(searchResultsAutocomplete, {
        options: ({ query, type }: { query: string; type: string }) => {
            const variables = {
                filter: {
                    type,
                },
                page: 0,
                query,
                size: 20,
            }
            return {
                variables,
            }
        },
    }),
    withLoading()
)(withPagination(Search, 'searchAutocomplete'))
