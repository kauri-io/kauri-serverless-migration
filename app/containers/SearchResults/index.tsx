import {
    Grid,
    Typography,
    InputBase,
    Tabs,
    Tab,
    Hidden,
    Menu,
    MenuItem,
    Theme,
} from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'
import SearchIcon from '@material-ui/icons/Search'
import { useState } from 'react'
import Head from 'next/head'
import { compose, graphql } from 'react-apollo'
import { searchResultsAutocomplete } from '../../queries/Search'
import DisplayResources from './DisplayResources'
import { connect } from 'react-redux'
import { routeChangeAction } from '../../lib/Epics/RouteChangeEpic'
import { debounce } from '../../lib/debounce'
import Chevron from '@material-ui/icons/ChevronLeft'

const debouncedRoute = debounce((query, routeChangeAction) => {
    if (query) {
        routeChangeAction(`/search-results?q=${query}`)
    }
}, 500)

const Search = ({ routeChangeAction, query, data: { searchAutocomplete } }) => {
    const [tab, setTab] = useState(0)

    const [createAnchorEl, setCreateAnchorEl] = useState<
        HTMLLIElement | HTMLButtonElement | null
    >(null)

    const isCreateMenuOpen = Boolean(createAnchorEl)

    const openDropdown = (
        event: React.MouseEvent<HTMLLIElement | HTMLButtonElement>
    ) => {
        setCreateAnchorEl(event.currentTarget)
    }

    const closeDropDown = () => {
        setCreateAnchorEl(null)
    }

    const useStyles = makeStyles((theme: Theme) => ({
        container: {},
        header: {
            height: 200,
            width: '100%',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            flexDirection: 'column',
            backgroundColor: 'white',
            background: `url(/static/images/HomepageBannerSVG.svg) center center no-repeat`,
        },
        heading: {
            margin: theme.spacing(1),
        },
        inputInput: {
            padding: theme.spacing(1, 2),
        },
        inputRoot: {
            color: 'inherit',
            width: '100%',
            paddingRight: theme.spacing(2),
        },
        searchClass: {
            alignItems: 'center',
            backgroundColor: theme.palette.background.default,
            borderRadius: theme.shape.borderRadius,
            display: 'flex',
            margin: theme.spacing(1),
            width: '90%',
            [theme.breakpoints.up('sm')]: {
                marginLeft: theme.spacing(3),
                width: 'auto',
            },
        },
        searchIconClass: {
            alignItems: 'center',
            display: 'flex',
            height: '100%',
            justifyContent: 'center',
            cursor: 'pointer',
            marginRight: theme.spacing(2),
        },
        dropdown: {
            margin: theme.spacing(2, 2, 0, 2),
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'space-between',
            cursor: 'pointer',
            '& > svg': {
                transition: 'all 0.2s',
                transform: createAnchorEl ? 'rotate(90deg)' : 'rotate(-90deg)',
            },
        },
        fullWidth: {
            width: '100%',
        },
    }))

    const classes = useStyles()

    const getDropdownString = tab => {
        switch (tab) {
            case 0:
                return `Articles (${(searchAutocomplete &&
                    searchAutocomplete.totalElementsBreakdown.ARTICLE) ||
                    0})`
            case 1:
                return `Collections (${(searchAutocomplete &&
                    searchAutocomplete.totalElementsBreakdown.COLLECTION) ||
                    0})`
            case 2:
                return `Communities (${(searchAutocomplete &&
                    searchAutocomplete.totalElementsBreakdown.COMMUNITY) ||
                    0})`
            case 3:
                return `Users (${(searchAutocomplete &&
                    searchAutocomplete.totalElementsBreakdown.USER) ||
                    0})`
        }
    }

    return (
        <Grid className={classes.container}>
            <Head>
                <title>{query.q} Kauri Search</title>
            </Head>
            <Grid className={classes.header}>
                <Typography className={classes.heading} variant="h5">
                    Search Results
                </Typography>
                <Typography variant="subtitle2">
                    {searchAutocomplete && searchAutocomplete.totalElements}{' '}
                    results
                </Typography>
                <div className={classes.searchClass}>
                    <InputBase
                        placeholder="Searchâ€¦"
                        onChange={e =>
                            debouncedRoute(e.target.value, routeChangeAction)
                        }
                        classes={{
                            input: classes.inputInput,
                            root: classes.inputRoot,
                        }}
                        defaultValue={query.q}
                    />
                    <div className={classes.searchIconClass}>
                        <SearchIcon />
                    </div>
                </div>
            </Grid>

            <Grid>
                <Hidden mdUp={true}>
                    <Typography
                        className={classes.dropdown}
                        onClick={openDropdown}
                    >
                        {getDropdownString(tab)} <Chevron />
                    </Typography>
                    <Menu
                        variant="selectedMenu"
                        open={isCreateMenuOpen}
                        anchorEl={createAnchorEl}
                        onClose={closeDropDown}
                        PaperProps={{
                            style: {
                                width: '100%',
                            },
                        }}
                    >
                        <MenuItem
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .ARTICLE
                            }
                            onClick={() => {
                                closeDropDown()
                                setTab(0)
                            }}
                        >
                            {getDropdownString(0)}
                        </MenuItem>
                        <MenuItem
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .COLLECTION
                            }
                            onClick={() => {
                                closeDropDown()
                                setTab(1)
                            }}
                        >
                            {getDropdownString(1)}
                        </MenuItem>
                        <MenuItem
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .COMMUNITY
                            }
                            onClick={() => {
                                closeDropDown()
                                setTab(2)
                            }}
                        >
                            {getDropdownString(2)}
                        </MenuItem>
                        <MenuItem
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown.USER
                            }
                            onClick={() => {
                                closeDropDown()
                                setTab(3)
                            }}
                        >
                            {getDropdownString(3)}
                        </MenuItem>
                    </Menu>
                </Hidden>
                <Hidden smDown={true}>
                    <Tabs
                        TabIndicatorProps={{ style: { height: 3 } }}
                        indicatorColor="primary"
                        centered={true}
                        value={tab}
                        onChange={(_e, tab) => setTab(tab)}
                    >
                        <Tab
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .ARTICLE
                            }
                            label={getDropdownString(0)}
                        />
                        <Tab
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .COLLECTION
                            }
                            label={getDropdownString(1)}
                        />
                        <Tab
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown
                                    .COMMUNITY
                            }
                            label={getDropdownString(2)}
                        />
                        <Tab
                            disabled={
                                searchAutocomplete &&
                                !searchAutocomplete.totalElementsBreakdown.USER
                            }
                            label={getDropdownString(3)}
                        />
                    </Tabs>
                </Hidden>
                {tab === 0 && (
                    <Grid>
                        <DisplayResources type="ARTICLE" query={query.q} />
                    </Grid>
                )}
                {tab === 1 && (
                    <Grid>
                        <DisplayResources type="COLLECTION" query={query.q} />
                    </Grid>
                )}
                {tab === 2 && (
                    <Grid>
                        <DisplayResources type="COMMUNITY" query={query.q} />
                    </Grid>
                )}
                {tab === 3 && (
                    <Grid>
                        <DisplayResources type="USER" query={query.q} />
                    </Grid>
                )}
            </Grid>
        </Grid>
    )
}

export default compose(
    connect(
        () => {},
        {
            routeChangeAction,
        }
    ),
    graphql(searchResultsAutocomplete, {
        options: ({
            query,
        }: {
            viewedSearchCategory: string | null
            query: {
                q: string
            }
        }) => {
            const variables = {
                page: 0,
                query: query.q,
                size: 1,
            }
            return {
                fetchPolicy: 'no-cache',
                variables,
            }
        },
    })
)(Search)
