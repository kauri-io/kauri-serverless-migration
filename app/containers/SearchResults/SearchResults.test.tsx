import SearchResults from './index'
import { mountWithRedux } from '../../setupTests'
import { MockedProvider } from 'react-apollo/test-utils'
describe('containers/SearchResults', () => {
    it('should match snapshot', () => {
        const props = {
            query: {
                q: 'eth',
            },
            router: {},
        }
        const wrapper = mountWithRedux(
            <MockedProvider>
                <SearchResults {...props} />
            </MockedProvider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
