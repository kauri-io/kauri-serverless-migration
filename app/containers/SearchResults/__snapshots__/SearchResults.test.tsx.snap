// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/SearchResults should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <MockedProvider
      addTypename={true}
    >
      <ApolloProvider
        client={
          ApolloClient {
            "__operations_cache__": Map {},
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "clearStoreCallbacks": Array [],
            "clientAwareness": Object {},
            "defaultOptions": Object {},
            "disableNetworkFetches": false,
            "link": ApolloLink {
              "request": [Function],
            },
            "mutate": [Function],
            "query": [Function],
            "queryDeduplication": true,
            "reFetchObservableQueries": [Function],
            "resetStore": [Function],
            "resetStoreCallbacks": Array [],
            "ssrMode": false,
            "store": DataStore {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "version": "2.4.13",
            "watchQuery": [Function],
          }
        }
      >
        <SearchResults
          query={
            Object {
              "q": "eth",
            }
          }
          router={Object {}}
        >
          <section>
            <SearchResults__ArticlesHeader>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "SearchResults__ArticlesHeader-sc-3rzbew-0",
                      "isStatic": false,
                      "lastClassName": "gMlgbs",
                      "rules": Array [
                        "background-color:",
                        [Function],
                        ";width:100%;display:flex;align-items:center;justify-content:center;flex-direction:column;color:",
                        [Function],
                        ";padding:",
                        [Function],
                        "px;padding-bottom:",
                        [Function],
                        "px;",
                      ],
                    },
                    "displayName": "SearchResults__ArticlesHeader",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "SearchResults__ArticlesHeader-sc-3rzbew-0",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="SearchResults__ArticlesHeader-sc-3rzbew-0 gMlgbs"
                >
                  <Typography__styledComponent
                    color="white"
                  >
                    <StyledComponent
                      color="white"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__styledComponent-sc-36xkf6-5",
                            "isStatic": false,
                            "lastClassName": "iLJkHN",
                            "rules": Array [
                              "color:",
                              [Function],
                              ";margin:0px;font-weight:",
                              "500",
                              ";",
                              ";font-size:",
                              "28",
                              "px;text-transform:",
                              [Function],
                              ";text-align:",
                              [Function],
                              ";:hover{color:",
                              [Function],
                              ";cursor:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "Typography__styledComponent",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="Typography__styledComponent-sc-36xkf6-5 iLJkHN"
                        color="white"
                      >
                        Search
                      </span>
                    </StyledComponent>
                  </Typography__styledComponent>
                  <Typography__component>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__component-sc-36xkf6-4",
                            "isStatic": false,
                            "lastClassName": "kdkRvh",
                            "rules": Array [
                              "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                              [Function],
                              "px;line-height:18px;color:",
                              [Function],
                              ";",
                              ";",
                              [Function],
                              ";",
                            ],
                          },
                          "displayName": "Typography__component",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__component-sc-36xkf6-4",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="Typography__component-sc-36xkf6-4 kdkRvh"
                      >
                        Loading results
                      </span>
                    </StyledComponent>
                  </Typography__component>
                  <withApollo(Connect(Complete))
                    query={
                      Object {
                        "q": "eth",
                      }
                    }
                    router={Object {}}
                    setSearchCategory={[Function]}
                    setSearchResults={[Function]}
                  >
                    <ApolloConsumer>
                      <Connect(Complete)
                        client={
                          ApolloClient {
                            "__operations_cache__": Map {},
                            "cache": InMemoryCache {
                              "addTypename": true,
                              "cacheKeyRoot": KeyTrie {
                                "weakness": true,
                              },
                              "config": Object {
                                "addTypename": true,
                                "dataIdFromObject": [Function],
                                "fragmentMatcher": HeuristicFragmentMatcher {},
                                "freezeResults": false,
                                "resultCaching": true,
                              },
                              "data": DepTrackingCache {
                                "data": Object {},
                                "depend": [Function],
                              },
                              "maybeBroadcastWatch": [Function],
                              "optimisticData": DepTrackingCache {
                                "data": Object {},
                                "depend": [Function],
                              },
                              "silenceBroadcast": false,
                              "storeReader": StoreReader {
                                "executeSelectionSet": [Function],
                                "executeStoreQuery": [Function],
                                "executeSubSelectedArray": [Function],
                                "freezeResults": false,
                              },
                              "storeWriter": StoreWriter {},
                              "typenameDocumentCache": Map {},
                              "watches": Set {},
                            },
                            "clearStoreCallbacks": Array [],
                            "clientAwareness": Object {},
                            "defaultOptions": Object {},
                            "disableNetworkFetches": false,
                            "link": ApolloLink {
                              "request": [Function],
                            },
                            "mutate": [Function],
                            "query": [Function],
                            "queryDeduplication": true,
                            "reFetchObservableQueries": [Function],
                            "resetStore": [Function],
                            "resetStoreCallbacks": Array [],
                            "ssrMode": false,
                            "store": DataStore {
                              "cache": InMemoryCache {
                                "addTypename": true,
                                "cacheKeyRoot": KeyTrie {
                                  "weakness": true,
                                },
                                "config": Object {
                                  "addTypename": true,
                                  "dataIdFromObject": [Function],
                                  "fragmentMatcher": HeuristicFragmentMatcher {},
                                  "freezeResults": false,
                                  "resultCaching": true,
                                },
                                "data": DepTrackingCache {
                                  "data": Object {},
                                  "depend": [Function],
                                },
                                "maybeBroadcastWatch": [Function],
                                "optimisticData": DepTrackingCache {
                                  "data": Object {},
                                  "depend": [Function],
                                },
                                "silenceBroadcast": false,
                                "storeReader": StoreReader {
                                  "executeSelectionSet": [Function],
                                  "executeStoreQuery": [Function],
                                  "executeSubSelectedArray": [Function],
                                  "freezeResults": false,
                                },
                                "storeWriter": StoreWriter {},
                                "typenameDocumentCache": Map {},
                                "watches": Set {},
                              },
                            },
                            "version": "2.4.13",
                            "watchQuery": [Function],
                          }
                        }
                        query={
                          Object {
                            "q": "eth",
                          }
                        }
                        router={Object {}}
                        setSearchCategory={[Function]}
                        setSearchResults={[Function]}
                      >
                        <Complete
                          client={
                            ApolloClient {
                              "__operations_cache__": Map {},
                              "cache": InMemoryCache {
                                "addTypename": true,
                                "cacheKeyRoot": KeyTrie {
                                  "weakness": true,
                                },
                                "config": Object {
                                  "addTypename": true,
                                  "dataIdFromObject": [Function],
                                  "fragmentMatcher": HeuristicFragmentMatcher {},
                                  "freezeResults": false,
                                  "resultCaching": true,
                                },
                                "data": DepTrackingCache {
                                  "data": Object {},
                                  "depend": [Function],
                                },
                                "maybeBroadcastWatch": [Function],
                                "optimisticData": DepTrackingCache {
                                  "data": Object {},
                                  "depend": [Function],
                                },
                                "silenceBroadcast": false,
                                "storeReader": StoreReader {
                                  "executeSelectionSet": [Function],
                                  "executeStoreQuery": [Function],
                                  "executeSubSelectedArray": [Function],
                                  "freezeResults": false,
                                },
                                "storeWriter": StoreWriter {},
                                "typenameDocumentCache": Map {},
                                "watches": Set {},
                              },
                              "clearStoreCallbacks": Array [],
                              "clientAwareness": Object {},
                              "defaultOptions": Object {},
                              "disableNetworkFetches": false,
                              "link": ApolloLink {
                                "request": [Function],
                              },
                              "mutate": [Function],
                              "query": [Function],
                              "queryDeduplication": true,
                              "reFetchObservableQueries": [Function],
                              "resetStore": [Function],
                              "resetStoreCallbacks": Array [],
                              "ssrMode": false,
                              "store": DataStore {
                                "cache": InMemoryCache {
                                  "addTypename": true,
                                  "cacheKeyRoot": KeyTrie {
                                    "weakness": true,
                                  },
                                  "config": Object {
                                    "addTypename": true,
                                    "dataIdFromObject": [Function],
                                    "fragmentMatcher": HeuristicFragmentMatcher {},
                                    "freezeResults": false,
                                    "resultCaching": true,
                                  },
                                  "data": DepTrackingCache {
                                    "data": Object {},
                                    "depend": [Function],
                                  },
                                  "maybeBroadcastWatch": [Function],
                                  "optimisticData": DepTrackingCache {
                                    "data": Object {},
                                    "depend": [Function],
                                  },
                                  "silenceBroadcast": false,
                                  "storeReader": StoreReader {
                                    "executeSelectionSet": [Function],
                                    "executeStoreQuery": [Function],
                                    "executeSubSelectedArray": [Function],
                                    "freezeResults": false,
                                  },
                                  "storeWriter": StoreWriter {},
                                  "typenameDocumentCache": Map {},
                                  "watches": Set {},
                                },
                              },
                              "version": "2.4.13",
                              "watchQuery": [Function],
                            }
                          }
                          query={
                            Object {
                              "q": "eth",
                            }
                          }
                          routeChangeAction={[Function]}
                          router={Object {}}
                          setSearchCategory={[Function]}
                          setSearchResults={[Function]}
                        >
                          <ResourceSearch__SearchWrapper
                            className="global-search-wrapper"
                          >
                            <StyledComponent
                              className="global-search-wrapper"
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "ResourceSearch__SearchWrapper-sc-7ioqrv-1",
                                    "isStatic": true,
                                    "lastClassName": "hwZepJ",
                                    "rules": Array [
                                      "width:300px;margin-bottom:64px;margin-top:19px;display:grid;position:relative;border-radius:4px;&:hover{& > input{border:1px solid #209b86;}}& > .certain-category-icon{position:absolute;top:11px;right:9px;}",
                                    ],
                                  },
                                  "displayName": "ResourceSearch__SearchWrapper",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "ResourceSearch__SearchWrapper-sc-7ioqrv-1",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <div
                                className="global-search-wrapper ResourceSearch__SearchWrapper-sc-7ioqrv-1 hwZepJ"
                              >
                                <ResourceSearch__SearchInput
                                  onChange={[Function]}
                                  value="eth"
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "ResourceSearch__SearchInput-sc-7ioqrv-0",
                                          "isStatic": false,
                                          "lastClassName": "fTtWOd",
                                          "rules": Array [
                                            "border:1px solid white;border-radius:4px;background:transparent;height:40px;outline:none;padding:0 ",
                                            [Function],
                                            "px;",
                                          ],
                                        },
                                        "displayName": "ResourceSearch__SearchInput",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "ResourceSearch__SearchInput-sc-7ioqrv-0",
                                        "target": "input",
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    onChange={[Function]}
                                    value="eth"
                                  >
                                    <input
                                      className="ResourceSearch__SearchInput-sc-7ioqrv-0 fTtWOd"
                                      onChange={[Function]}
                                      value="eth"
                                    />
                                  </StyledComponent>
                                </ResourceSearch__SearchInput>
                                <SearchSVG>
                                  <div
                                    className="certain-category-icon"
                                  >
                                    <svg
                                      fill="white"
                                      height="18"
                                      viewBox="0 0 24 24"
                                      width="18"
                                    >
                                      <path
                                        clipRule="evenodd"
                                        d="M11 4C7.13401 4 4 7.13401 4 11C4 14.866 7.13401 18 11 18C14.866 18 18 14.866 18 11C18 7.13401 14.866 4 11 4ZM2 11C2 6.02944 6.02944 2 11 2C15.9706 2 20 6.02944 20 11C20 15.9706 15.9706 20 11 20C6.02944 20 2 15.9706 2 11Z"
                                        fill="white"
                                        fillRule="evenodd"
                                      />
                                      <path
                                        clipRule="evenodd"
                                        d="M15.9429 15.9429C16.3334 15.5524 16.9666 15.5524 17.3571 15.9429L21.7071 20.2929C22.0976 20.6834 22.0976 21.3166 21.7071 21.7071C21.3166 22.0977 20.6834 22.0977 20.2929 21.7071L15.9429 17.3571C15.5524 16.9666 15.5524 16.3334 15.9429 15.9429Z"
                                        fill="white"
                                        fillRule="evenodd"
                                      />
                                    </svg>
                                  </div>
                                </SearchSVG>
                              </div>
                            </StyledComponent>
                          </ResourceSearch__SearchWrapper>
                        </Complete>
                      </Connect(Complete)>
                    </ApolloConsumer>
                  </withApollo(Connect(Complete))>
                </div>
              </StyledComponent>
            </SearchResults__ArticlesHeader>
            <ResourceResults
              loading={true}
              query="eth"
              setSearchCategory={[Function]}
              totalElementsBreakdown={
                Object {
                  "ARTICLE": 0,
                  "COLLECTION": 0,
                  "COMMENT": 0,
                  "COMMUNITY": 0,
                  "CURATED_LIST": 0,
                  "REQUEST": 0,
                  "USER": 0,
                }
              }
            >
              <LoadingComponent>
                <style
                  type="text/css"
                >
                  
        .centered-wrapper{
          display: flex;
          flex: 1;
          align-items: center;
          justify-content: center;
        }
        .lds-ellipsis {
          display: inline-block;
          position: relative;
          width: 64px;
          height: 64px;
        }
        .lds-ellipsis div {
          position: absolute;
          top: 27px;
          width: 11px;
          height: 11px;
          border-radius: 50%;
          background: #0BA986;
          animation-timing-function: cubic-bezier(0, 1, 1, 0);
        }
        .lds-ellipsis div:nth-child(1) {
          left: 6px;
          animation: lds-ellipsis1 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(2) {
          left: 6px;
          animation: lds-ellipsis2 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(3) {
          left: 26px;
          animation: lds-ellipsis2 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(4) {
          left: 45px;
          animation: lds-ellipsis3 0.6s infinite;
        }
        @keyframes lds-ellipsis1 {
          0% {
            transform: scale(0);
          }
          100% {
            transform: scale(1);
          }
        }
        @keyframes lds-ellipsis3 {
          0% {
            transform: scale(1);
          }
          100% {
            transform: scale(0);
          }
        }
        @keyframes lds-ellipsis2 {
          0% {
            transform: translate(0, 0);
          }
          100% {
            transform: translate(19px, 0);
          }
        }
        
                </style>
                <div
                  className="centered-wrapper"
                >
                  <div
                    className="lds-ellipsis"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </div>
              </LoadingComponent>
            </ResourceResults>
          </section>
        </SearchResults>
      </ApolloProvider>
    </MockedProvider>
  </ThemeProvider>
</Provider>
`;
